src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -142,6 +142,56 @@ private static TimeZoneInfo GetLocalTimeZoneCore()
             return Utc;
         }
 
+        private static TimeSpan GetCacheLocalUtcOffset(DateTime dateTime, TimeZoneInfoOptions flags)
+        {
+                CachedData cachedData = s_cachedData;
+                return cachedData.Local.GetUtcOffset(dateTime, flags, cachedData);
+        }
+
+        private static TimeSpan? _localUtcOffset;
+        private static object _localUtcOffsetLock = new();
+        private static Thread? _loadAndroidTZData;
+        // Shortcut for TimeZoneInfo.Local.GetUtcOffset
+        // On Android, loading AndroidTZData while obtaining cachedData.Local is expensive for startup.
+        // We introduce a fast result for GetLocalUtcOffset that relies on the date time offset being
+        // passed into monovm_initialize(_preparsed) from Java in seconds as LOCAL_DATE_TIME_OFFSET.
+        // However, to handle timezone changes during the app lifetime, AndroidTZData needs to be loaded.
+        // The fast path is initially used, and we start a background thread to get cachedData.Local
+        internal static TimeSpan GetLocalUtcOffset(DateTime dateTime, TimeZoneInfoOptions flags)
+        {
+            if (_localUtcOffset != null) // The background thread finished, the cache is loaded.
+                return GetCacheLocalUtcOffset(dateTime, flags);
+
+            if (_localUtcOffset == null && _loadAndroidTZData == null) // The cache isn't loaded and no background thread has been created
+            {
+                lock (_localUtcOffsetLock)
+                {
+                    // GetLocalUtcOffset may be called multiple times before a cache is loaded and a background thread is running,
+                    // once the lock is available, check for a cache and background thread.
+                    if (_localUtcOffset != null)
+                        return GetCacheLocalUtcOffset(dateTime, flags);
+
+                    if (_loadAndroidTZData == null)
+                    {
+                        _loadAndroidTZData = new Thread(() => {
+                            _localUtcOffset = GetCacheLocalUtcOffset(dateTime, flags);
+                            Thread.Sleep(1000);
+                        });
+                        _loadAndroidTZData.IsBackground = true;
+                        _loadAndroidTZData.Start();
```

Comment:
@grendello what are your thoughts here on starting a `Thread`?

Reply:
The thread idea was my suggestion and I think it's going to work quite OK, provided it's a bare thread as above and not part of a threadpool or TPL in general. It shouldn't hurt startup performance too much, although what I'd do would be to let it sleep **first** for 1 second and then start loading the TZ. The idea is that it should start after the main startup sequence of the app, and 1s should be enough time.

Reply:
If 1s isn't enough time, is this going to make launching the app even slower?